### PR TITLE
Housekeeping

### PR DESCRIPTION
**Pull request by ZeroChaos-**
Some basic housekeeping items.

- [x] Fix LED killing to not commit hari kari

- [x] Speed boot by not starting unneeded services (stealth and speed)

- [x] Start dnsmasq when needed, stop when not needed

- [x] Intentionally randomize mac address in attack mode

- [x] Don't pass default route when in arming mode

- [x] Don't try to start openvpn

- [x] More POSIX and less bash for speed

- [x] Arming Mode works

- [x] Attack Mode works

- [x]  Extensions work

- [x] Off Mode works

- [x] Switching mode works

- [x] Ensure attack mode comes up with no networking no matter the previous state

### CODE REVIEW

**Comment by ZeroChaos-** [comment]
*needs pkill*

File: usr/bin/NETMODE [hunk 1]
```diff
@@ -1,34 +1,39 @@
...
 
-function show_usage() {
+show_usage() {
     echo "Usage: $0 [DHCP_CLIENT|DHCP_SERVER|AUTO]"
     echo ""
 }
 
-function shut_it_all_down() {
+shut_it_all_down() {
     #we are restarting, possibly with new config, or reversing dhcp roles
```
Comment (foxtrot) [inline usr/bin/NETMODE]: Please drop comments

File: usr/bin/cleanup [hunk 2]
```diff
@@ -1,12 +1,6 @@
-#!/bin/bash
-
-# List of directories to clean
-DIRS=(
-    ~/.ssh
-    /root/loot
-)
+#!/bin/sh
```
Comment (foxtrot) [inline usr/bin/cleanup:1]: Doesn't make sense to get rid of the array here, it's easier to expand in the future.

File: usr/bin/execute_payload [hunk 3]
```diff
@@ -1,34 +1,54 @@
...
-    sed -i 's/\r//g' $payload
-    bash -C "${payload}" &> /dev/null
-    ;;
-  *)
-    /usr/bin/LED FAIL
-    ;;
+    "payload.py")
+        python "${payload}" > /dev/null 2>&1
+        ;;
+    "payload.php")
```
Comment (foxtrot) [inline usr/bin/execute_payload]: PHP support can be dropped :)

**Review by foxtrot — CHANGES_REQUESTED** [review]

File: usr/bin/NETMODE (hunk 1 above)
Comment (ZeroChaos-) [reply usr/bin/NETMODE]: comments don't slow the code and explain to others the intent of the code. on principle, no.

File: usr/bin/cleanup (hunk 2 above)
Comment (ZeroChaos-) [reply usr/bin/cleanup:1]: arrays aren't supported in posix shell. granted, this isn't the slowest of the scripts, but it still runs twice as fast in posix shell than it does in bash

File: usr/bin/execute_payload (hunk 3 above)
Comment (ZeroChaos-) [reply usr/bin/execute_payload]: pre-existing, but sure.

File: usr/bin/NETMODE (hunk 1 above)
Comment (foxtrot) [reply usr/bin/NETMODE]: I know comments don't slow execution. The function is simple enough to not need a comment.

File: usr/bin/cleanup (hunk 2 above)
Comment (foxtrot) [reply usr/bin/cleanup:1]: Fair enough, I actually think it's unlikely the list would be expanded in the future anyway, I agree it's worth getting rid of considering the speed advantage then.

Comment (ZeroChaos-) [reply usr/bin/cleanup]: my concern was that this function looks like it does too much if you don't consider all the options (like switching mode and coming from an unknown state).  I wanted it clear why the function was so kill happy

File: usr/bin/NETMODE (hunk 1 above)
Comment (ZeroChaos-) [reply usr/bin/NETMODE]: my concern was that this function looks like it does too much if you don't consider all the options (like switching mode and coming from an unknown state). I wanted it clear why the function was so kill happy

**Review by foxtrot — APPROVED** [review]

**Comment by foxtrot** [comment]
Thanks!

**Pull request merged**